radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -185,12 +188,9 @@ private RadixApplicationAPI(
 		this.universe = universe;
 		this.actionStores = atomMappers.stream().collect(Collectors.toMap(
 			AtomToExecutedActionsMapper::actionClass,
-			m -> new ActionStore<>(ledger.getAtomStore(), m)
-		));
-		this.applicationStores = particleReducers.stream().collect(Collectors.toMap(
-			ParticleReducer::stateClass,
-			r -> new ApplicationStore<>(ledger.getParticleStore(), r)
+			m -> m
 		));
+		this.applicationStores = particleReducers.stream().collect(Collectors.toMap(ParticleReducer::stateClass, r -> r));
```

Comment:
Can there only be one reducer per state class?

Reply:
yes